Airflow/dags/task_2.py

```diff
@@ -0,0 +1,96 @@
+'''
+Task 2 DAG for BigData lab homework
+'''
+
+from airflow import DAG
+from airflow.operators.bash import BashOperator
+from airflow.operators.python import PythonVirtualenvOperator
+from airflow.utils.dates import days_ago
+
+from datetime import timedelta
+
+
+# Set default tasks parameters
+default_args = {
+        'owner': 'airflow',
+        'start_date': days_ago(2),
+        'retries': 5,
+        'retry_delay': timedelta(seconds=20)
+        }
+
+# Initiate DAG
+
+with DAG('Homework-task2',
+    default_args=default_args,
+    description='csv row counting and avro saving') as dag:
+
+    dag.doc_md = '''
+    Counts rows for input csv table,
+    report row_count to stdout,
+    saves to avro format (with overwrite).
+    '''
+
+    # TODO: replace by Airflow variables
+
+    CSV_URL = 'https://www.stats.govt.nz/assets/Uploads/Greenhouse-gas-emissions-by-region-industry-and-household/Greenhouse-gas-emissions-by-region-industry-and-household-year-ended-2018/Download-data/greenhouse-gas-emissions-by-region-industry-and-household-year-ended-2018-csv.csv'
+    DATA_FOLDER = 'data'
+
+    
+    download = BashOperator(
+        task_id = 'download',
+        bash_command = f'mkdir -p $AIRFLOW_HOME/data; curl {CSV_URL} > $AIRFLOW_HOME/data/downloaded_file.csv; echo file downloaded'
+    )
+
+    row_count = BashOperator(
+        task_id='row_count',
+        bash_command = 'wc -l < $AIRFLOW_HOME/data/downloaded_file.csv'
+    )
+
+
+    def convert_fastavro():
+        from fastavro import writer, parse_schema
```

Comment:
Yes for more programs, but when work with PythonVirtualenvOperator imports of specific modules should be run inside function, which is called inside PythonVirtualenvOperator task, because  the task expected to runs in separate isolated venv.

See reference: https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html#pythonvirtualenvoperator